CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+# SLB
+#
+# Copyright Michael Park, 2017
+# Copyright Agustin Berge, 2017
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE.md or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.0)
+
+project(SLB CXX)
+
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  option(WITH_TESTS "Compile with unit tests." ON)
+  set(HAVE_TESTS ${WITH_TESTS})
+else()
+  option(SLB_WITH_TESTS "Compile with unit tests." ON)
+  set(HAVE_TESTS ${SLB_WITH_TESTS})
+endif()
+
+# Build
+add_library(_slb INTERFACE)
+target_include_directories(_slb INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+set_target_properties(_slb PROPERTIES
+  EXPORT_NAME SLB)
+
+add_library(SLB ALIAS _slb)
+
+# Test
+if (HAVE_TESTS)
+  enable_testing()
+  add_subdirectory(test)
+endif()
+
+# Config
+include(CMakePackageConfigHelpers)
+
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/slb-config.cmake.in [[
+# SLB
+#
+# Copyright Michael Park, 2017
+# Copyright Agustin Berge, 2017
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE.md or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# - Config file for SLB
+#
+#  SLB_INCLUDE_DIRS - include directories
+#  SLB_LIBRARIES    - libraries to link against
+#
+# The following `IMPORTED` target is also defined:
+#
+#  SLB
+
+@PACKAGE_INIT@
+
+include(${CMAKE_CURRENT_LIST_DIR}/slb-targets.cmake)
+
+get_target_property(SLB_INCLUDE_DIRS SLB INTERFACE_INCLUDE_DIRECTORIES)
+set_and_check(SLB_INCLUDE_DIRS "${SLB_INCLUDE_DIRS}")
+set(SLB_LIBRARIES SLB)
+]])
+
+configure_package_config_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/slb-config.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/slb-config.cmake
+  INSTALL_DESTINATION lib/cmake/slb
+  NO_CHECK_REQUIRED_COMPONENTS_MACRO)
+
+# Install
+set(_headers
+  slb/detail/config.hpp
+  slb/detail/lib.hpp
+  slb/type_traits.hpp)
+foreach (_header ${_headers})
+  get_filename_component(_directory ${_header} DIRECTORY)
+  install(
+    FILES include/${_header}
+    DESTINATION include/${_directory})
+endforeach()
+
+install(TARGETS _slb EXPORT _targets)
+install(EXPORT _targets
+  DESTINATION lib/cmake/slb
```

Comment:
I think we should flip the `FILE` and `DESTINATION` to stay consistent with other orderings.

Reply:
`FILE` here is the output filename within `DESTINATION`, not an input (the input is the `_targets` export set). Should I still flip it?

Reply:
Oh, I see. That was naive of me. Let's leave it.